tests/subprocesstest.py

```diff
@@ -57,8 +57,8 @@ def test_case(self):
 class SubprocessTestCase(unittest.TestCase):
     def _full_method_name(self):
         test = getattr(self, self._testMethodName)
-        modpath = test.__module__
-        clsname = self.__class__.__name__
+        modpath = test.__self__.__class__.__module__
```

Comment:
This change fixes an issue where a test method not overridden by a child class (so the parent method is called) results in an invalid `_full_method_name`.

For example the test method `tests.contrib.gevent.test_patch.TestGeventPatch.test_patch_before_import` which MROs to `tests.contrib.patch.PatchTestCase.Base.test_patch_before_import` (since it was not overridden) resulted in `_full_method_name() == 'tests.contrib.patch.TestGeventPatch.test_patch_before_import'` which is invalid.